marda_extractors_api/__init__.py

```diff
@@ -207,6 +241,32 @@ def _parse_python_arg(arg: str):
 
         return function_tree, args, kwargs
 
+    def _execute_python_venv(self, entry_command: str, setup: str):
+        data = None
+
+        with multiprocessing.managers.SharedMemoryManager() as shmm:
+            shm = shmm.SharedMemory(size=1024 * 1024 * 1024)
+            py_cmd = (
+                r'print("Launching extractor"); import '
+                + setup
+                + "; import pickle; import multiprocessing.shared_memory; shm = multiprocessing.shared_memory.SharedMemory(name="
+                + r"'"
+                + r"{}".format(shm.name)
+                + r"'"
+                + "); data = pickle.dumps("
+                + r"{}".format(entry_command)
+                + '); shm.buf[:len(data)] = data; print("Done!")'
+            )
+            if not self.venv_dir:
```

Comment:
Maybe `assert self.venv_dir is not None` instead.

Reply:
i don't like asserts for logic as they can be skipped with `python -O <file>`, but sure this could be made clearer